src/gtfs.rs

```diff
@@ -66,6 +72,11 @@ impl TryFrom<RawGtfs> for Gtfs {
             (*fare_rules.entry(f.fare_id.clone()).or_default()).push(f);
         }
 
+        let mut fare_products = HashMap::<String, Vec<FareProduct>>::new();
```

Comment:
This seems to be the minimal required annotation. I have no strong opinion about being explicit vs. being conscise
```suggestion
        let mut fare_products = HashMap::<_, Vec<_>>::new();
```